File: ChatGPT/python/output/output_23a.py
# ChatGPT-generated solution will go here.
import itertools
from collections import defaultdict

def main():
    with open("input.txt") as f:
        edges = [line.strip().split('-') for line in f if line.strip()]

    graph = defaultdict(set)
    for a, b in edges:
        graph[a].add(b)
        graph[b].add(a)

    count = 0
    nodes = list(graph.keys())

    for a, b, c in itertools.combinations(nodes, 3):
        if b in graph[a] and c in graph[a] and a in graph[b] and c in graph[b] and a in graph[c] and b in graph[c]:
            if a.startswith('t') or b.startswith('t') or c.startswith('t'):
                count += 1

    print(count)

if __name__ == "__main__":
    main()

File: ChatGPT/clojure/output/output_23a.clj
;; ChatGPT-generated solution will go here.
(ns output_23a
  (:require [clojure.string :as str]
            [clojure.set :as set]))

(defn build-graph [edges]
  (reduce (fn [g [a b]]
            (-> g
                (update a (fnil conj #{}) b)
                (update b (fnil conj #{}) a)))
          {}
          edges))

(defn find-triangles [graph]
  (let [nodes (keys graph)]
    (for [a nodes
          b (graph a)
          :when (str/compare a b) ; prevent duplicate sets
          c (graph b)
          :when (and (not= a c)
                     (contains? (graph a) c)
                     (compare b c))] ; ensure unique triangle
      (->> [a b c] sort vec))))

(defn part [lines]
  (let [edges (map #(str/split % #"-") lines)
        graph (build-graph edges)
        triangles (find-triangles graph)]
    (->> triangles
         (filter #(some (fn [n] (str/starts-with? n "t")) %))
         count)))

(defn -main [& args]
  (let [filename (first args)
        lines (str/split-lines (slurp filename))]
    (println (part lines))))
